app/db/repository/authenticator_repository.rb

```diff
@@ -8,11 +8,18 @@ def initialize(data_object:, resource_repository: ::Resource, logger: Rails.logg
       end
 
       def find_all(type:, account:)
+        identifier = Sequel.function(:identifier, :resource_id)
+
         @resource_repository.where(
           Sequel.like(
             :resource_id,
             "#{account}:webservice:conjur/#{type}/%"
           )
+        ).where(
+          Sequel.like(
+            identifier,
+            %r{conjur/(authn(?:-[^/]+)?(?:/[^/]+)?)$}
```

Comment:
It would be worth adding a comment here explaining how this regex prevents the status resources from being included in the result. (No forward slashes are allowed after `authn`).

Reply:
Also, given this isn't an urgent change, would should improve this re-use beyond just copying the regex between this location and https://github.com/cyberark/conjur/blob/master/app/domain/authentication/installed_authenticators.rb#L31.

For example, it should probably be a constant in one class or the other, and just reference in the other location:

```rb
# installed_authenticators.rb

...
class InstalledAuthenticators
    ...
    AUTHN_RESOURCE_PREFIX = "conjur/authn-"
    AUTHN_STATUS_FILTER = %r{conjur/(authn(?:-[^/]+)?(?:/[^/]+)?)$}
    ...
    # filter out nested status webservice
    .map { |id| id[AUTHN_STATUS_FILTER, 1] } 
    ...
```

```rb
# authenticator_repository.rb
...
...
Sequel.like(
  identifier,
  # filter out nested status webservice
  Authentication::InstalledAuthenticators::AUTHN_STATUS_FILTER
)
...
```

Reply:
@mFelgate Final comment here, and perhaps this is the most important. This needs an integration test that actually runs this query and verifies the result, because I don't believe this is going to work as-is.

The `Sequel.like` method expects a SQL  LIKE statement, and not a regex:
https://www.rubydoc.info/github/jeremyevans/sequel/Sequel%2FSQL%2FStringExpression.like